connectors/byoc.py

```diff
@@ -668,71 +616,99 @@ async def prepare_docs(self, data_provider, pipeline=None, filtering=None):
 
     async def prepare(self, config):
         """Prepares the connector, given a configuration
-
-        If the connector id and the service is in the config, we want to
+        If the connector id and the service type is in the config, we want to
         populate the service type and then sets the default configuration.
-
-        Returns the source class.
         """
         configured_connector_id = config.get("connector_id", "")
-        configured_service_type = config.get("service_type", "")
+        if self.id != configured_connector_id:
+            return
 
-        if self.id == configured_connector_id and self.service_type is None:
+        if self.service_type is not None and not self.configuration.is_empty():
+            return
+
+        doc = {}
+        if self.service_type is None:
+            configured_service_type = config.get("service_type", "")
             if not configured_service_type:
                 logger.error(
                     f"Service type is not configured for connector {configured_connector_id}"
                 )
                 raise ServiceTypeNotConfiguredError("Service type is not configured.")
-            self.service_type = configured_service_type
-            logger.debug(f"Populated service type for connector {self.id}")
-
-        service_type = self.service_type
-        if service_type not in config["sources"]:
-            raise ServiceTypeNotSupportedError(service_type)
+            self.service_type = doc["service_type"] = configured_service_type
+            logger.debug(
+                f"Populated service type {configured_service_type} for connector {self.id}"
+            )
 
-        fqn = config["sources"][service_type]
-        try:
-            source_klass = get_source_klass(fqn)
+        if self.configuration.is_empty():
+            if configured_service_type not in config["sources"]:
+                raise ServiceTypeNotSupportedError(configured_service_type)
+            fqn = config["sources"][configured_service_type]
```

Comment:
```suggestion
            source_name = config["sources"][configured_service_type]
```

I think `fqn` is a bit cryptic, if you've never heard it before.

Reply:
One example is `connectors.sources.mysql:MySqlDataSource`, so `fqn` is more accurate than `source_name`. 

I think it's just a convention to use abbreviation in Python and you need to get used to it (I was not initially as well)

Reply:
yeah it's "Fully Qualified Name" -- maybe we could use `class_path` to make it easier to understand